Feature/multi module presentation

kotrasatheesh-dev opened the pull request.

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Added Multi Module presentation

kotrasatheesh-dev committed:
Feat:presentation screen2

kotrasatheesh-dev committed:
Fix:some changes made

kotrasatheesh-dev committed:
Delete presentation/src/main/java/com/example/presentation/navigation/characters/AllCharacters.kt

kotrasatheesh-dev committed:
Fix:files some issues fixed

kotrasatheesh-dev committed:
Fix:files some issues fixed

kotrasatheesh-dev committed:
Fix:files some issues fixed

kotrasatheesh-dev committed:
Fix:files some issues fixed

kotrasatheesh-dev committed:
Removed unused dependency from libs.versions.toml and build.gadle

kotrasatheesh-dev committed:
Fix:files added dependencys

kotrasatheesh-dev committed:
Fix:files added dependencys

kotrasatheesh-dev committed:
Fix:files added dependencys

ds-griddynamics commented on presentation/src/main/java/com/example/presentation/navigation/CharacterAppBar.kt:
```diff
@@ -0,0 +1,32 @@
+        title: String,
+        modifier: Modifier = Modifier
+    ) {
+        Row(
```
Do a have a look `CenterAlignedTopAppBar`

ds-griddynamics commented on presentation/src/main/java/com/example/presentation/navigation/MainScreen.kt:
```diff
@@ -0,0 +1,33 @@
+import com.example.presentation.navigation.viewmodel.CharactersViewModel
+
+@Composable
+fun MainScreen(
```
Try to have a look on `State Hoisting` in compose

ds-griddynamics commented on presentation/src/main/java/com/example/presentation/navigation/characters/CharacterListView.kt:
```diff
@@ -0,0 +1,32 @@
+        horizontalArrangement = Arrangement.spacedBy(20.dp),
+        verticalArrangement = Arrangement.spacedBy(20.dp),
+    ) {
+        items(charactersList?.size ?: 0) { index ->
```
null safe required ?

ds-griddynamics commented on presentation/src/main/java/com/example/presentation/navigation/characters/CharacterListView.kt:
```diff
@@ -0,0 +1,32 @@
+
+
+@Composable
+fun CharactersList(
```
Assuming this shows Grid List, you can rename the fun to something similar

ds-griddynamics commented on presentation/src/main/java/com/example/presentation/navigation/viewmodel/CharactersViewModel.kt:
```diff
@@ -0,0 +1,39 @@
+        viewModelScope.launch {
+            _charactersState.value = UiState.Loading
+            val characters = charactersUseCase.invokeCharacters()
+            if (characters.isNotEmpty()) {
+                _charactersState.value = UiState.Success(characters)
+            } else {
+                _charactersState.value = UiState.Error("No characters found")
```
this checks can be moved to use case

kotrasatheesh-dev replied:
here The domain module (where CharacterUseCase resides) is independent of the presentation module (where UiState resides). The ViewModel takes the raw data from the CharacterUseCase and maps it to the UiState, which avoids any circular dependency between layers

ds-griddynamics reviewed (changes requested):
Great work, some minor changes requested

kotrasatheesh-dev committed:
Fix:PR comments resolved
AllCharacters
CharacterAppBar
CharactersGridList
CharactersViewModel
MainActivity
MainScreen
MainScreenContainer

kotrasatheesh-dev committed:
Fix:PR comments resolved
AllCharacters
Build_gradle
